### PR TITLE
RFC #157 : Add feature meta tag to web-platform-tests meta data

### PR DESCRIPTION
**Pull request by jcscottiii**
[Rendered](https://github.com/web-platform-tests/rfcs/blob/Add-feature-meta-tag-to-web-platform-tests-meta-data/rfcs/feature-meta-tag-to-web-platform-tests-meta-data.md)

<details>
  <summary>Expand to see diagram mentioned in Implementation Details section</summary>
  
  image
</details>

### CODE REVIEW

**Comment by jcscottiii** [comment]
@jgraham I have added the suggestions discussed in the WPT infra meeting yesterday.

Any thoughts @jgraham @gsnedders?

**Review by Ms2ger — APPROVED** [review]
I have no particular objection to this.

We will need documentation in the wpt repo that explains clearly that this is optional; please take a look where you'd like to put this.

File: rfcs/feature-meta-tag-to-web-platform-tests-meta-data.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+# Alternatives considered
+
+One alternative considered was to add specific web platform tests to each feature-set definition instead. This would be similar to how BCD data is included in the definition ([Example](https://github.com/web-platform-dx/feature-set/blob/c9314c5126e74f4e1fecd56202774a845c66f9b4/feature-group-definitions/subgrid.yml#L9-L10)).  However, this would cause a developer experience where WPT authors would need to land changes in both WPT and feature-set. Given that test names in WPT change more frequently than feature group definitions, it could add unnecessary toil to maintain the connection.
+
+# Implementation Details
+
+There are two steps that need to happen:  
+1. Annotate the META.yml files with the "feature" key that corresponds to the feature-set entry. This can be done gradually and not all at once.
+2. Create a wpt script that generates a Feature manifest. Similar to the [SPEC_MANIFEST.json](https://github.com/web-platform-tests/wpt/pull/40655). For more details, check the diagram attached to the pull request for this RFC.
```
Comment (jgraham) [inline rfcs/feature-meta-tag-to-web-platform-tests-meta-data.md:77]: AFAICT the prior art here all uses _either_ META.yml files _or_ in-test metadata. Obviously it's not impossible to combine those things, but the diagram on the PR suggests quite a lot of engineering overhead for this feature.

I wonder if the easier way to start isn't to use the wpt-metadata repo for this kind of labeling task instead, like we do for Interop labels? I'd generally be in favour of having a discussion about how to unify the various labels we have (e.g. there would be significant advantages of having the interop labelling in-tree).

**Review by jgraham — COMMENTED** [review]
This is OK, but I am slightly concerned about the approach suggested being quite complicated and adding yet another special case for metadata that isn't quite like the spec metadata or the bug metadata or the interop labelling metadata.

File: rfcs/feature-meta-tag-to-web-platform-tests-meta-data.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+# Changes for WPT Contributors
+
+The metadata tag is not mandatory. WPT contributions will not be blocked if contributors do not add the metadata tag to the META.yml files or test files.
+
+# Populating and maintaining the metadata tags
+
+As the [feature-set](https://github.com/web-platform-dx/feature-set) repository is populated with feature-set definitions, feature-set contributors will begin populating the metadata in WPT.
```
[the remark this answers is not in the document]
Comment (jcscottiii) [reply rfcs/feature-meta-tag-to-web-platform-tests-meta-data.md:85]: @jgraham in previous experiments with metadata, was there a way to accomplish this?

**Comment by jcscottiii** [comment]
@jgraham @domenic @Ms2ger To address the unnecessary review concerns and incorporate the feedback from the latest WPT meeting, I created a new RFC: https://github.com/web-platform-tests/rfcs/pull/163

**Comment by jcscottiii** [comment]
Close in favor of #163

**Pull request closed without merging**